sphinx/util/inspect.py

```diff
@@ -389,12 +389,23 @@ def is_builtin_class_method(obj: Any, attr_name: str) -> bool:
     Why this function needed? CPython implements int.__init__ by Descriptor
     but PyPy implements it by pure Python code.
     """
-    classes = [c for c in inspect.getmro(obj) if attr_name in c.__dict__]
-    cls = classes[0] if classes else object
+    try:
+        mro = inspect.getmro(obj)
```

Comment:
off topic: We have many codes that touches `__mro__` directly. Is it better to replace them with `inspect.getmro()`?

Reply:
No idea. The implementation of `getmro` on my machine is just `.__mro__`.

Reply:
Okay. I'll keep them as is.